Sources/Networking/Core/Upload/UploadTask+State.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  UploadTask+State.swift
+//  
+//
+//  Created by Tony Ngo on 12.06.2023.
+//
+
+import Foundation
+
+extension UploadTask {
+    /// The upload task's state.
+    public struct State {
+        /// Number of bytes sent.
+        public let sentBytes: Int64
+
+        /// Number of bytes expected to send.
+        public let totalBytes: Int64
+
+        /// An error produced by the task.
+        public var error: Error?
+
+        /// A response produced by the task.
+        public var response: Response?
+
+        /// The internal state of the `URLSessionTask`.
+        let taskState: URLSessionTask.State
```

Comment:
why this is not public? can it be private in other case?

Reply:
It's an internal implementation detail that I didn't want to expose to the client. It's not private so it can be accessible in potential tests.